pipeline/utils/github_utils.py

```diff
@@ -0,0 +1,165 @@
+    curr_dir and the remote tree  have conflicting contents for files with the
+    same path, the new tree will use the content of curr_dir.
+
+    Returns:
+        New tree object as a dictionary, if curr_dir had files
+        Otherwise, returns None
+
+    Raises:
+        HTTPError: An HTTP request used to construct the tree failed
+
+    """
+    if orig_tree_sha:
+        orig_tree = _exec_request(
+            requests.get, base_url + 'git/trees/{0}'.format(orig_tree_sha))
+    tree_els = []
+    for subdir in _get_immediate_subdirs(curr_dir):
+        subtree_sha = None
+        if orig_tree_sha:
+            subtree_sha = _get_subtree_sha(orig_tree, subdir)
+        subtree = _extend_git_tree(os.path.join(curr_dir, subdir), username,
+                                   password, base_url, subtree_sha)
+        # Add the subtree to tree_els iff subdir contains files
+        if subtree:
+            tree_els.append({'path': subdir, 'mode': _TREE_MODE,
+                             'type': 'tree', 'sha': subtree['sha']})
+    for exec_or_file in _get_immediate_files(curr_dir):
+        with open(os.path.join(curr_dir, exec_or_file), 'r') as f:
+            path_content = f.read()
+        tree_els.append({'path': exec_or_file,
+                         'mode': _get_file_mode(os.path.join(curr_dir,
+                                                             exec_or_file)),
+                         'type': 'blob', 'content': path_content})
+    # Git doesn't allow empty trees
+    if tree_els:
+        data_dict = {'tree': tree_els}
+        if orig_tree_sha:
+            data_dict['base_tree'] = orig_tree_sha
+        return _exec_request(requests.post, base_url + 'git/trees',
+                             json.dumps(data_dict), username, password)
+
+
+def push_dir_github(output_dir, username, password, owner, repo, branch,
+                    message='Automated commit from artman'):
+    """Push all content in output_dir to the given GitHub repo branch.
+
+    The effect of running the task is not exactly the same as that of a
+    forced git push: this won't change or delete files that are missing
+    in the local repo but are present in the remote repo.
+
+    Returns:
+        None
+
+    Raises:
+        ValueError: output_dir was empty
+        HTTPError: An HTTP request used to construct the Git tree failed
+
+    """
+    base_url = '{0}/repos/{1}/{2}/'.format(_API_URL, owner, repo)
+    # get the sha of the latest commit on this branch
+    branch_item = _exec_request(
+        requests.get,
+        base_url + 'branches/{0}'.format(branch))
+    commit_sha = branch_item['commit']['sha']
+    orig_tree_sha = branch_item['commit']['commit']['tree']['sha']
+    root_tree = _extend_git_tree(output_dir, username, password,
+                                 base_url, orig_tree_sha)
+    if not root_tree:
+        raise ValueError("Cannot push empty folder {0}".format(output_dir))
```

Comment:
nit: prefer single quotes for strings for consistency with the rest of the codebase